$OpenBSD: patch-src_runtime_Config_generic-openbsd,v 1.2 2017/12/25 07:04:45 jca Exp $

clang-5.0.0 doesn't pass -export-dynamic to the linker, instead it
passes a bogus -e export-dynamic to ld(1). Just use -Wl,--export-dynamic.

Don't try to guess (wrong) with clang. Just assume we have pie

Index: src/runtime/Config.generic-openbsd
--- src/runtime/Config.generic-openbsd.orig
+++ src/runtime/Config.generic-openbsd
@@ -9,16 +9,12 @@
 # provided with absolutely no warranty. See the COPYING and CREDITS
 # files for more information.
 
-LINKFLAGS += -export-dynamic -Wl,-z,wxneeded
+LINKFLAGS += -Wl,--export-dynamic -Wl,-z,wxneeded
 OS_LIBS += -lutil
 
-# If the compiler recognizes a -nopie flag then we're building on a
-# PIE-by-default system, so disable PIE.
-ifneq ($(shell $(CC) -dumpspecs 2>/dev/null | grep -e -nopie),)
 CFLAGS += -fno-pie
 LINKFLAGS += -nopie
 LDFLAGS += -nopie
-endif
 
 ifdef LISP_FEATURE_SB_THREAD
 CFLAGS += -pthread
